coralproject/talk
# [CORL-2331] Searchable Sites List

## What does this PR do?
This PR addresses [CORL-2331](https://vmproduct.atlassian.net/browse/CORL-2331) by adding a site search by name. This is to support cases, such as SBNation, that include many sites, to make it easier for someone to go through the list and find a specific site they're looking for. These changes add site search components to the admin that are then used in the stories route. These include a `SiteSearchListContainer` that displays search results, the `SiteSearchTextField`, and the `SiteFilterOption`, which shows each search option. The `SiteFilterOption` component was already being used by the stories route before.   The idea is that in the future, these components could also be used in the moderation route in place of its current `SiteSelector` dropdown, as well as potentially in other places where a site search would be useful.

## What changes to the GraphQL/Database Schema does this PR introduce?
This adds a query to the text index for sites for the text search by name.

## How do I test this PR?
You’ll need to have multiple sites (at least 10 is ideal) added. A few of the sites should have at least one story. Then, navigate to the stories route. Search within the new “Search by site name” input. 

Test selecting different sites changes the input value. Test selecting different sites correctly updates the stories shown . Test that selecting “All sites” shows all stories. Test that clicking outside of the site search closes the site search result list if it’s open. Test that the search result list shows both “All sites” and “No sites were found with that search” when no site is found (any feedback on what messaging is shown in this case is welcomed!).
 
## How do we deploy this PR?
Upon deploy, I believe that we will need to add a text index for the sites query.
```
use coral;
db.sites.createIndex({ tenantID: 1, "$**": "text", createdAt: -1 });
```
